ludwig/encoders/generic_encoders.py

```diff
@@ -65,7 +65,7 @@ class DenseEncoder(Encoder):
     def __init__(
         self,
         input_size,
-        layers=None,
+        fc_layers=None,
```

Comment:
@metamath1 Thank you for proposing this change.  Do you think it would potentially cause backward compatibility issues if other users already passed `layers` to the constructor?  Thank you. /cc @arnavgarg1

Reply:
🙇

Reply:
@metamath1 Actually, I am a bit confused about this PR - what's the intended scope and reason for the change? 

I misunderstood the change - this may not require any backward incompatibility changes since you're modifying the argument in the encoder class as opposed to the schema directly.

Reply:
> @alexsherstinsky Yes, that is right.
> 
> We have a file to add support for backward compatibility here: https://github.com/ludwig-ai/ludwig/blob/ea890d99e72fbc32fad8aae18070da9969ddb422/ludwig/utils/backward_compatibility.py
> 
> And an associated test file https://github.com/ludwig-ai/ludwig/blob/ea890d99e72fbc32fad8aae18070da9969ddb422/tests/ludwig/utils/test_backward_compatibility.py to add a test to make sure it works.
> 
> @metamath1 Are you able to add backward compatibility support and an associated test for it?

Here's the setup for creating multiple Linear layers in a dense encoder

```
config = {
    'preprocessing': {
        'split': {
            'type': 'fixed',
            'column': 'split'
        }
    },
    'input_features': [
        {
            'name': 'image',
            'type': 'vector',
            'encoder': {
                'type': 'dense',
                # This part doesn't work.
		# This part is ignored and always creates a layer with an output size of 256.
		'fc_layers': [
                    {'output_size': 64},
                    {'output_size': 32}
                ]
            }
        }
    ],
    'output_features': [
        {'name': 'label', 'type': 'category'}
    ],
    'trainer': {'epochs': 1, 'batch_size': 64, 'eval_batch_size': 128}
}
```

However, the `fc_layers` setting does not work.
This is because the argument to the DenseEncoder is named `layers`, 
but the config dictionary object stores the setting as `fc_layers`.
The settings in the user's `fc_layers` are not being passed to the DenseEncoder.

Reply:
This really needs to be fixed.

For retrocompatibility we can have a couple lines fo code that internally rename `layers` to `fc_layers` and warn the user of the change and that it is deprecated.

Also, if it was not working anyway, there's not a lot of damage in beckward incompatibility.

Finally, I'm surprised by the fact that the wrong keyword was not captured by either the config validar or the dereferencing of the kwargs, the should have been a check somewhere.